src/traffic_server/traffic_server.cc

```diff
@@ -452,6 +453,33 @@ class MemoryLimit : public Continuation
   struct rusage _usage;
 };
 
+/** Gate the emission of the "Traffic Server is fuly initialized" log message.
+ *
+ * This message is intended to be helpful to users who want to know that
+ * Traffic Server is not just running but has become fully initialized and is
+ * ready to optimize traffic. This is in contrast to the "traffic server is
+ * running" message which can be printed before either of these conditions.
+ *
+ * This function is called on each initialization state transition. Currently,
+ * the two state transitions of interest are:
+ *
+ * 1. The cache is initialized.
```

Comment:
Should this say "If enabled, the cache is initialized".

Reply:
Can you avoid cache initialization? I found that even if http.cache.http was disabled the cache was still initialized.

I can look into this further.

Reply:
I can believe it, it's trafficserver.

Reply:
I enabled the "cache" debug tag and ran the deny0 test which disables the cache. The cache does indeed get initialized with this output indicating cache initialization:

```
[Jun 24 17:23:28.592] traffic_server DEBUG: <Cache.cc:3068 (ink_cache_init)> (cache_init) proxy.config.cache.ram_cache.size = -1 = 0Mb
[Jun 24 17:23:28.592] traffic_server DEBUG: <Cache.cc:3076 (ink_cache_init)> (cache_init) proxy.config.cache.limits.http.max_alts = 5
[Jun 24 17:23:28.592] traffic_server DEBUG: <Cache.cc:3080 (ink_cache_init)> (cache_init) cache_config_ram_cache_cutoff = 4194304 = 4Mb
[Jun 24 17:23:28.592] traffic_server DEBUG: <Cache.cc:3083 (ink_cache_init)> (cache_init) proxy.config.cache.permit.pinning = 0
[Jun 24 17:23:28.592] traffic_server DEBUG: <Cache.cc:3086 (ink_cache_init)> (cache_init) proxy.config.cache.dir.sync_frequency = 60
[Jun 24 17:23:28.592] traffic_server DEBUG: <Cache.cc:3089 (ink_cache_init)> (cache_init) proxy.config.cache.select_alternate = 1
[Jun 24 17:23:28.592] traffic_server DEBUG: <Cache.cc:3093 (ink_cache_init)> (cache_init) proxy.config.cache.max_doc_size = 0 = 0Mb
[Jun 24 17:23:28.592] traffic_server DEBUG: <Cache.cc:3096 (ink_cache_init)> (cache_init) proxy.config.cache.mutex_retry_delay = 2ms
[Jun 24 17:23:28.592] traffic_server DEBUG: <Cache.cc:3099 (ink_cache_init)> (cache_init) proxy.config.cache.read_while_writer.max_retries = 10
[Jun 24 17:23:28.592] traffic_server DEBUG: <Cache.cc:3102 (ink_cache_init)> (cache_init) proxy.config.cache.read_while_writer_retry.delay = 50ms
[Jun 24 17:23:28.592] traffic_server DEBUG: <Cache.cc:3105 (ink_cache_init)> (cache_init) proxy.config.cache.hit_evacuate_percent = 0
[Jun 24 17:23:28.592] traffic_server DEBUG: <Cache.cc:3108 (ink_cache_init)> (cache_init) proxy.config.cache.hit_evacuate_size_limit = 0
[Jun 24 17:23:28.592] traffic_server DEBUG: <Cache.cc:3121 (ink_cache_init)> (cache_init) proxy.config.cache.max_disk_errors = 5
[Jun 24 17:23:28.593] traffic_server DEBUG: <Cache.cc:3124 (ink_cache_init)> (cache_init) proxy.config.cache.agg_write_backlog = 5242880
[Jun 24 17:23:28.593] traffic_server DEBUG: <Cache.cc:3127 (ink_cache_init)> (cache_init) proxy.config.cache.enable_checksum = 0
[Jun 24 17:23:28.593] traffic_server DEBUG: <Cache.cc:3130 (ink_cache_init)> (cache_init) proxy.config.cache.alt_rewrite_max_size = 4096
[Jun 24 17:23:28.593] traffic_server DEBUG: <Cache.cc:3135 (ink_cache_init)> (cache_init) proxy.config.cache.enable_read_while_writer = 1
[Jun 24 17:23:28.593] traffic_server DEBUG: <Store.cc:331 (read_config)> (cache_init) Store::read_config, fd = -1, "/tmp/sbdelay/deny0/ts/config/storage.config"
[Jun 24 17:23:28.593] traffic_server DEBUG: <Store.cc:364 (read_config)> (cache_init) Store::read_config: "storage"
[Jun 24 17:23:28.593] traffic_server DEBUG: <Store.cc:403 (read_config)> (cache_init) Store::read_config - ns = new Span; ns->init("/tmp/sbdelay/deny0/ts/storage",268435456), forced volume=-1
[Jun 24 17:23:28.593] traffic_server DEBUG: <Store.cc:594 (init)> (cache_init) initialized span '/tmp/sbdelay/deny0/ts/storage'
[Jun 24 17:23:28.593] traffic_server DEBUG: <Store.cc:596 (init)> (cache_init) hw_sector_size=4096, size=268435456, blocks=32768, disk_id=64514/197593927, file_pathname=0
[Jun 24 17:23:28.757] traffic_server DEBUG: <Cache.cc:688 (start_internal)> (cache_hosting) Disk: 0:/tmp/sbdelay/deny0/ts/storage/cache.db, blocks: 32767
[Jun 24 17:23:28.768] [ET_NET 9] DEBUG: <Cache.cc:2666 (cplist_reconfigure)> (cache_hosting) blocks = 32766
[Jun 24 17:23:28.769] [ET_NET 9] DEBUG: <Cache.cc:810 (diskInitialized)> (cache_hosting) Disk: 0:/tmp/sbdelay/deny0/ts/storage/cache.db: Vol Blocks: 1: Free space: 0
[Jun 24 17:23:28.769] [ET_NET 9] DEBUG: <Cache.cc:812 (diskInitialized)> (cache_hosting)        Vol: 0 Size: 32766
[Jun 24 17:23:28.769] [ET_NET 9] DEBUG: <Cache.cc:816 (diskInitialized)> (cache_hosting)        Block No: 0 Size: 32766 Free: 0
[Jun 24 17:23:28.769] [ET_NET 9] DEBUG: <Cache.cc:2065 (open)> (cache_init) Cache::open - proxy.config.cache.min_average_object_size = 8000
[Jun 24 17:23:28.778] [ET_NET 9] DEBUG: <Cache.cc:1256 (init)> (cache_init) Vol /tmp/sbdelay/deny0/ts/storage/cache.db 16384:32766: allocating 352256 directory bytes for a 268419072 byte volume (0.131234%)
[Jun 24 17:23:28.782] [ET_NET 10] DEBUG: <CacheHosting.cc:433 (Init)> (cache_hosting) Host Record: 0x7f8f68028838, Volume: 0, size: 32766
[Jun 24 17:23:28.782] [ET_NET 10] DEBUG: <Cache.cc:1885 (build_vol_hash_table)> (cache_init) build_vol_hash_table index 0 mapped to 0 requested 32707 got 32707
[Jun 24 17:23:28.782] [ET_NET 10] DEBUG: <Cache.cc:863 (cacheInitialized)> (cache_init) CacheProcessor::cacheInitialized - theCache, total_size = 32766 = 255 MB
[Jun 24 17:23:28.782] [ET_NET 10] DEBUG: <Cache.cc:892 (cacheInitialized)> (cache_init) CacheProcessor::cacheInitialized - caches_ready=0x9, gnvol=1
[Jun 24 17:23:28.782] [ET_NET 10] DEBUG: <Cache.cc:911 (cacheInitialized)> (cache_init) CacheProcessor::cacheInitialized - cache_config_ram_cache_size == AUTO_SIZE_RAM_CACHE
[Jun 24 17:23:28.782] [ET_NET 10] DEBUG: <Cache.cc:919 (cacheInitialized)> (cache_init) CacheProcessor::cacheInitialized - ram_cache_bytes = 352256 = 0Mb
[Jun 24 17:23:28.782] [ET_NET 10] DEBUG: <Cache.cc:925 (cacheInitialized)> (cache_init) CacheProcessor::cacheInitialized - total_cache_bytes = 268066816 = 255Mb

```

Thanks for your thoughtful questions.